plugins/modules/packaging/os/yum_versionlock.py

```diff
@@ -128,15 +129,15 @@ def main():
                 if module.check_mode:
                     changed = True
                     continue
-                changed = yum_v.ensure_state(single_pkg, command)
+        changed = yum_v.ensure_state(packages, command)
```

Comment:
Two comments:
1. This ignores check mode :)
2. How about using the above loop to compile a list of packages which actually have to be changed, and then passing only these packages on to `ensure_state`?

(Same for below.)

Reply:
This fixes check mode, https://github.com/ansible-collections/community.general/pull/1912/commits/58e580646310768c70e018b4d070736be836dbc2.

Reply:
It does, but it also passes a lot of packages that do not need to be passed.

Reply:
I guess this only passes needed packages, https://github.com/ansible-collections/community.general/pull/1912/commits/f0947cbd27635e76ba5d6e46117b23faa8a2e3dc. Could you please check?